### PR TITLE
(OraklNode) Refactor common types

### PR DESCRIPTION
**Pull request by nick-bisonai**
# Description

common package to store common types and redis keys

Fixes # (issue)

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## Checklist before requesting a review

- [ ] I have performed a self-review of my code.
- [ ] If it is a core feature, I have added thorough tests.

## Deployment

- [ ] Should publish npm package
- [ ] Should publish Docker image

### CODE REVIEW

File: node/pkg/common/keys/keys.go [hunk 1]
```diff
@@ -0,0 +1,29 @@
+package keys
+
+import (
+	"strconv"
+)
+
+func LatestFeedDataKey(feedID int32) string {
+	return "latestFeedData:" + strconv.Itoa(int(feedID))
```
Comment (Intizar-T) [inline node/pkg/common/keys/keys.go:8]: if feedID is already passed as an int32, i wonder why are casting it to int here?

Comment (nick-bisonai) [reply node/pkg/common/keys/keys.go:8]: In Go, the strconv.Itoa function is designed to convert an int type to a string representation. It does not have built-in support for converting int32 or other integer types directly.

😞

File: node/pkg/aggregator/types.go [hunk 2]
```diff
@@ -48,18 +45,7 @@ type Proof struct {
 	Proof    []byte `json:"proofs"`
 }
 
-type PgsLocalAggregate struct {
-	ConfigID  int32     `db:"configId"`
-	Value     int64     `db:"value"`
-	Timestamp time.Time `db:"timestamp"`
-}
```
Comment (Intizar-T) [inline node/pkg/aggregator/types.go:55]: I dont see a replacement for this struct. Does that mean it was not being used anywhere else?

Comment (nick-bisonai) [reply node/pkg/aggregator/types.go:55]: My first intention was that the data format could've been different between rdb and pgsql. But later on it ended up as a same structure. You can think that this type has been merged into `types.LocalAggregate`

it was referenced in utils as followed before
```
func GetLatestLocalAggregateFromPgs(ctx context.Context, configId int32) (PgsLocalAggregate, error) {
	return db.QueryRow[PgsLocalAggregate](ctx, SelectLatestLocalAggregateQuery, map[string]any{"config_id": configId})
}
```

But now it just uses `LocalAggregate`

**Review by Intizar-T — APPROVED** [review]
lgtm! 

A minor suggestion, the word id is referenced as "Id" in some instances and "ID" in others. Would be great to keep them consistent! but it's good as it is as well 👍

**Comment by nick-bisonai** [comment]
> lgtm!
> 
> A minor suggestion, the word id is referenced as "Id" in some instances and "ID" in others. Would be great to keep them consistent! but it's good as it is as well 👍

Totally agreed, will make an update

**Pull request merged**